src/java/com/phenix/pct/PCTASBroker.java

```diff
@@ -0,0 +1,401 @@
+     * 
+     * @param append Boolean
+     */
+    public void setServerLogFileAppend(boolean append) {
+        this.serverLogFileAppend = append;
+    }
+
+    /**
+     * Working directory of servers. No default value.
+     * 
+     * @param workDir File
+     */
+    public void setWorkDir(File workDir) {
+        this.workDir = workDir;
+    }
+
+    /**
+     * Controlling name server. Default value is NS1
+     * 
+     * @param ns String
+     */
+    public void setNameServer(String ns) {
+        this.nameServer = ns;
+    }
+
+    /**
+     * Server settings
+     * 
+     * @param sp ServerProcess (customized version of PCTRun)
+     */
+    public void addConfiguredServer(ServerProcess sp) {
+        if (this.server == null) {
+            this.server = sp;
+        } else {
+            throw new BuildException("Only one server process...");
+        }
+    }
+
+    /**
+     * Do the work
+     * 
+     * @throws BuildException Something went wrong
+     */
+    @Override
+    public void execute() {
+        File propFile = null;
+
+        try {
+            checkAttributes();
+            // Choosing right properties file
+            if (file == null) {
+                propFile = new File(getDlcHome(), "properties/" + UBROKER_PROPERTIES);
+            } else {
+                propFile = getProject().resolveFile(file);
+            }
+            if (!propFile.exists())
+                throw new BuildException("Unable to find properties file "
+                        + propFile.getAbsolutePath());
+            // And do the work
+            writeDeltaFile();
+            Task task = getCmdLineMergeTask(propFile, tmpFile);
+            task.execute();
+        } catch (BuildException be) {
+            throw be;
+        } finally {
+            cleanup();
+        }
+    }
+
+    /**
+     * Write files to be used as a parameter for the MergeProp shell script.
+     * 
+     * @throws BuildException Something went wrong during write.
+     * 
+     */
+    private void writeDeltaFile() {
+        try (Writer w = new FileWriter(tmpFile); PrintWriter bw = new PrintWriter(w)) {
+            bw.println("[UBroker.AS." + name + "]");
+            if (!"delete".equalsIgnoreCase(action)) {
+                bw.println("appserviceNameList=" + name);
+                bw.println("registerNameServer=1");
+                bw.println("controllingNameServer=" + nameServer);
+                bw.println("registrationMode=Register-IP");
+                bw.println("operatingMode=" + operatingMode);
+                bw.println("autoStart=" + (autoStart ? "1" : "0"));
+                if (server != null) {
+                    if (!"".equals(server.getPropath())) {
+                        bw.print("PROPATH=");
+                        bw.println(server.getPropath());
+                    }
+
+                    // TODO Erm, this is crap... I should use something else to handle correctly quotes in command line
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1135)